server.php

```diff
@@ -65,9 +65,11 @@ function valet_support_xip_io($domain)
 $valetSitePath = null;
 
 foreach ($valetConfig['paths'] as $path) {
-    if (is_dir($path.'/'.$siteName)) {
-        $valetSitePath = $path.'/'.$siteName;
-
+    $domain = ($pos = strrpos($siteName, '.')) !== false
+                ? substr($siteName, $pos+1)
+                : null;
+    if (is_dir($path.'/'.$siteName) || is_dir($path.'/'.$domain)) {
```

Comment:
If you do not do the valet link on another app somewhere, everything works, as the ~/.valet/Sites is not added to valets config.json file.